### PR TITLE
feat: adds more descriptive subdomain errors

### LINKED ISSUE #207: Give a nicer error message back if a user runs subdomain command with their existing subdomain
I was expecting that I would need to run the new `subdomain` command with my workers.dev subdomain, to register it in the configuration. I did that and got this error back:

```sh
worker $ wrangler subdomain signalnerve
🐌  Registering your subdomain, signalnerve.workers.dev, this could take up to a minute.
Error: ErrorMessage { msg: "⛔ This account already has a registered subdomain. You can only register one subdomainper account. Your subdomain is signalnerve.workers.dev \n Status Code: 409 Conflict\n Msg: {\n  \"result\": null,\n  \"success\": false,\n  \"errors\": [\n    {\n      \"code\": 10036,\n      \"message\": \"workers.api.error.account_has_subdomain\"\n    }\n  ],\n  \"messages\": []\n}\n" }
```

It would be neat to validate the user's existing subdomain against the argument they passed in for the command: if they're the same, we can return something back like "Your domain is already registered!"

### PR DESCRIPTION
**Pull request by EverlastingBugstopper**
Warns users when attempting to add a subdomain w/o account_id in
wrangler.toml

Doesn't show failure when attempting to register the subdomain that
is already associated with their account

Fixes #207

### CODE REVIEW

File: src/commands/subdomain.rs [hunk 1]
```diff
@@ -83,32 +86,36 @@ pub fn subdomain(name: &str, user: &GlobalUser, project: &Project) -> Result<(),
         .body(sd_request)
         .send()?;
 
-    let msg;
+    let mut msg = None;
```
Comment (EverlastingBugstopper) [inline src/commands/subdomain.rs]: I'm not sure if this is the best way to do this, pointers much appreciated

Comment (EverlastingBugstopper) [reply src/commands/subdomain.rs]: nvm: figured out how to return early :)

File: src/commands/subdomain.rs [hunk 2]
```diff
@@ -89,12 +94,21 @@ pub fn subdomain(name: &str, user: &GlobalUser, project: &Project) -> Result<(),
...
             let sd = Subdomain::get(account_id, user)?;
-            msg = format!(
-                "⛔ This account already has a registered subdomain. You can only register one subdomain per account. Your subdomain is {}.workers.dev \n Status Code: {}\n Msg: {}",
-                sd,
-                res.status(),
-                res_text,
-            );
+            if sd == name {
+                println!(
+                    "{} Your subdomain {}.workers.dev has already been registered!",
```
Comment (ashleygwilliams) [inline src/commands/subdomain.rs]: can you say more about what this message is trying to communicate? on first glance i read it and it seemed as though it was supposed to be an error message, but the sparkles seemed confusing. i *think* you mean it as an informative message... let's figure out what exactly we're communicating here, would love to hear more about how you thought about it!

Comment (EverlastingBugstopper) [reply src/commands/subdomain.rs]: So this case is for when someone tries to register the subdomain they've already registered. I don't really think we need to return an error message here (which is the current behavior). Perhaps there's a better way to say this...

Comment (EverlastingBugstopper) [reply src/commands/subdomain.rs]: If that's not clear, this message is returned when avery@cloudflare.com has already registered avery.workers.dev and runs `wrangler subdomain avery`

Comment (signalnerve) [reply src/commands/subdomain.rs]: I would read this as "Your subdomain has already been registered _(by someone else)_" – is that the intent here? If not, should it be something like "You've already registered the subdomain xyz.workers.dev"?

Comment (signalnerve) [inline src/commands/subdomain.rs]: I would read this as "Your subdomain has already been registered (by someone else)" – is that the intent here? If not, should it be something like "You've already registered the subdomain xyz.workers.dev"?

Comment (EverlastingBugstopper) [reply src/commands/subdomain.rs]: Yeah so this case is specifically for when the user is trying to register the subdomain that they themselves have already registered. The latter makes sense to me

**Pull request merged**